### ISSUE TITLE
[BUG] AnimateSharedLayout should counteract animation of the target tree animation

### ISSUE THREAD

**Issue by paales** [framer/motion#849]
**1. Read the [FAQs](#faqs) 👇**
✅

**2. Describe the bug**

When animated a shared layout, it doesn't counter-animate one of the parent's animations in the target or source tree.

**3. IMPORTANT: Provide a CodeSandbox reproduction of the bug**

https://codesandbox.io/s/framer-motion-animatesharedlayout-app-store-demo-forked-n6x72?file=/src/Item.js

**4. Steps to reproduce**

1. Go to demo
2. Click on an image
3. Click on the overlay image
4. You now see that the 'transitioned' image still animates down

**5. Expected behavior**

I would expect the image to animate to its target and not animate down.

**Comment by rijk** [comment]
This appears to be fixed in 2.9.4: https://codesandbox.io/s/framer-motion-animatesharedlayout-app-store-demo-forked-dxhez?file=/src/Item.js

**Comment by paales** [comment]
Damn, used the wrong version in the sandbox.. There does however persist an inconsistency in the demo (which might be an implementation issue):

I'd like the animation to go to its final target directly, completely ignoring the animation of its parent. Currently the item animates to the position in the parent first and then animates to its final position. This of course looks pretty jarring.

So essentially I can't use x/y animations while using a shared layout element in that way. Any ideas how to work around this?

**Comment by rijk** [comment]
I'm sorry, I don't really understand what your desired outcome is.

**Comment by paales** [comment]
Thanks for taking your time! Let me try again.

I'd like to create a bottom drawer that animates from the bottom (`y:500` in the example) to its final position (`y:0` in the example). Meanwhile I'd like to animate the picture from the source location (the grid item in the example) to the final target location.

What it does:
It animates the picture from the grid to the `y:500` position and then animates the drawer and the image to `y:0`

What i'd like it to do:
It animates the picture from the grid to the `y:0` position while the drawer it's self animates from `y:500` to `y:0`

**Comment by rijk** [comment]
Ah, ok. Yeah then I think you'd need to split the elements up, so that the image starts at `y: 0`, and the rest (text etc) animates from `y: 500`.

**Comment by paales** [comment]
Hmm, feels kinda gross, but I think that should be doable 🤔
